### PR TITLE
Issue #238: update to CS 8.23

### LINKED ISSUE #238: upgrade to checkstyle 8.23
https://checkstyle.org/releasenotes.html#Release_8.23 , upgrade version at https://github.com/checkstyle/sonar-checkstyle/edit/master/pom.xml#L88

>New:
    new filter SuppressionSingleFilter. 
    OneStatementPerLine: add treatTryResourcesAsStatement property.
    MagicNumber: new proprty ignoreAnnotationElementDefaults to ignore magic numbers in annotation default values. 
    new check: InvalidJavadocPosition.

### PR DESCRIPTION
**Pull request by muhlba91**
fix #238 
javadoc_comment
magic_number
one_statement

singlesuppression enabled for line 5:
suppression

config:
one_statement_conf
javadoc_comment_conf
magic_number_conf
filter_conf

### CODE REVIEW

File: src/main/java/org/sonar/plugins/checkstyle/CheckstyleProfileImporter.java [hunk 1]
```diff
@@ -61,6 +61,7 @@
         "SuppressionCommentFilter",
         "SuppressWithNearbyCommentFilter",
         "SuppressionXpathSingleFilter",
+        "SuppressionSingleFilter",
```
Comment (rnveach) [inline src/main/java/org/sonar/plugins/checkstyle/CheckstyleProfileImporter.java]: Do we have to do anything special for a new filter?
I remember we have this weird dialog in sonar where we have to manually put in the filter's config.
Did you verify you can use this filter without issue?

Comment (rnveach) [reply src/main/java/org/sonar/plugins/checkstyle/CheckstyleProfileImporter.java]: Guess I am reacting too fast, picture came up showing its usage.

Comment (muhlba91) [reply src/main/java/org/sonar/plugins/checkstyle/CheckstyleProfileImporter.java]: yes, it's working. i uploaded the images.
here again for this particular filter.

before filter:
magic_number

filter:
filter_conf

after filter:
suppression
(line 5 got suppressed)

**Comment by romani** [comment]
Thanks a lot !

**Pull request merged**